### PR TITLE
Added to docs for Decimal class/constructor.

### PR DESCRIPTION
**Pull request by zslayton**
*Issue #, if available:* #365, #376

*Description of changes:*

* Adds a definition of the terms `exponent` and `coefficient` to `Decimal`'s class-level docs, along with examples.
* Adds TSDoc for `Decimal`'s public constructor.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/IonDecimal.ts [hunk 1]
```diff
@@ -17,7 +17,22 @@ import {_sign} from "./util";
 
 /**
  * This class provides the additional semantics necessary for decimal values.
- * Note that range of an exponent is limited to +/- 15 digits.
+ *
+ * Unlike IEEE-754 floating point representation, Decimal values can represent numeric values without loss of precision.
+ * They combine an arbitrarily large integer value (the "coefficient") with a second integer representing its scale
```
Comment (pbcornell) [inline src/IonDecimal.ts]: Most definitions of "scale" are "the number of digits to the right of the decimal point."  It's a bit different than exponent.

File: src/IonDecimal.ts [hunk 2]
```diff
@@ -17,7 +17,22 @@ import {_sign} from "./util";
...
+ * (the "exponent") using the following formula:
+ *
+ *     value = coefficient * (10 ^ exponent)
+ *
+ * Here are some examples:
+ *
+ *      coefficient     exponent        value
+ *             0           0                0
+ *           314          -2                3.14
+ *           -76           1              -76
```
Comment (pbcornell) [inline src/IonDecimal.ts]: something's off in this example:  `-76d1` would be `-760` (vs. `-76d0`, which would be `-76`)

File: src/IonDecimal.ts [hunk 3]
```diff
@@ -17,7 +17,22 @@ import {_sign} from "./util";
...
  * This class provides the additional semantics necessary for decimal values.
- * Note that range of an exponent is limited to +/- 15 digits.
+ *
+ * Unlike IEEE-754 floating point representation, Decimal values can represent numeric values without loss of precision.
+ * They combine an arbitrarily large integer value (the "coefficient") with a second integer representing its scale
+ * (the "exponent") using the following formula:
+ *
+ *     value = coefficient * (10 ^ exponent)
+ *
+ * Here are some examples:
```
Comment (pbcornell) [inline src/IonDecimal.ts]: maybe it would be helpful to have an example such as `0.00037` ?

File: src/IonDecimal.ts (hunk 1 above)
Comment (zslayton) [reply src/IonDecimal.ts]: I wasn't aware that 'scale' had an official meaning in this space, but that makes sense. I'll rephrase it.

File: src/IonDecimal.ts (hunk 2 above)
Comment (zslayton) [reply src/IonDecimal.ts]: Whoops, good catch!

File: src/IonDecimal.ts (hunk 3 above)
Comment (zslayton) [reply src/IonDecimal.ts]: Yep, added.

**Comment by zslayton** [comment]
Rebased from `master` and moved to #386

**Pull request closed without merging**